### DISCUSSION TITLE
On the problem of infinite for loops

### DISCUSSION THREAD

**Discussion by Sakura0001** [golangci/golangci-lint#4744] [Q&A] [answered]
I would like to ask a question. Currently, there is a problem with infinite continuity in the code under the business. I would like to ask if there is a Linter who can solve this problem. If you can answer this question, I would greatly appreciate it.

```go
for {
	req.PageIndex = &pageIndex
	curResp, err := ad_audit_hammurabi_core_i18n.RawCall.SearchWhiteListData(ctx, req)
	if err != nil {
		logs.CtxError(ctx, "error")
		continue
	}
	data = append(data, curResp.Datas...)
	if curResp.Count == nil || int64((*req.PageSize)*(*req.PageIndex)) > *curResp.Count{
		break
	}
	pageIndex++
}
```

**Comment by ldez** [comment, marked as answer]
>  I would like to ask if there is a Linter who can solve this problem. 

No, because it's impossible to evaluate statically this kind of problem.
Your snippet is an example: there is a break but it's impossible to know statically if the break condition is possible or not.

**Reply by Sakura0001** [reply]
Thank you for your answer, so can this question only be discovered through specific tests? Our tool is only suitable for static code detection, right?

**Reply by ldez** [reply]
golangci-lint is a static analysis tool.

To detect your problem, you can use tests and maybe fuzzing.
